src/interfaces.ts

```diff
@@ -121,7 +121,7 @@ export interface WorkerOptions extends WorkerSharedOptions {
   workerId?: string;
 }
 
-export interface WorkerPoolOptions extends WorkerSharedOptions {
+export interface WorkerPoolOptions extends WorkerOptions {
```

Comment:
This isn't valid; a worker pool is for managing multiple workers (a pool of workers); each worker needs its own ID, so specifying a workerId at this level would only work for a pool of size 1.

Reply:
(I don't want to open this can of worms; can we instead change the relevant place that uses WorkerPoolOptions that you think should use WorkerOptions instead?)

Reply:
> Going back after writing this, it's kinda all over the place.  As you read, just pick out the parts that make any sense :)

So I was working off the possibility that it was a typo to begin with, and the apparent usage of the options in JS. I realized the issues you mention afterwards.  I don't think there's a place to swap it out, and like I said the options seemed to be destructured in a way consitent with this change. (maybe not? see below)

I had put in the docs "only use with concurrency 1" but all that seems pretty dumb.  I agree this is a bad change.

I can't really think of a satisfying solution other than a config option that generates custom worker IDs.  

Overall, the default behavior serves *my* needs well, so I think probably just reject this change.  You had told me on discord that you can pass `workerId` at the top level, and you can in JS and it would work in a way consistent with this change, which is a little weird.  But I guess the more correct typing would be to revert the above and instead do:

```ts
export interface RunnerOptions extends WorkerOptions, WorkerPoolOptions {...}
```

Reply:
But...no matter how you expose `workerId` at the top level, it's still problematic like you said because it will break with concurrency > 1.  So I'd actually kinda prefer disallowing it at the top level and passing an optional generator function.  But it's not clear that anyone wants or needs that feature, so I think I still kinda vote to close this unless there's something I'm missing.